/** @addtogroup OceanFFTStandAlone */
/*@{*/

#ifndef MAINWINDOW_H
#define MAINWINDOW_H
#include "OpenGLWidget.h"
#include <QMainWindow>
//----------------------------------------------------------------------------------------------------------------------
namespace Ui {
    class MainWindow;
}
//----------------------------------------------------------------------------------------------------------------------
/// @brief A class to handle our window, GUI
//----------------------------------------------------------------------------------------------------------------------
//----------------------------------------------------------------------------------------------------------------------
class MainWindow : public QMainWindow
{
    Q_OBJECT
public:
    //----------------------------------------------------------------------------------------------------------------------
    /// @brief Constructor
    /// @param parent the parent widget associating with this widget
    //----------------------------------------------------------------------------------------------------------------------
    explicit MainWindow(QWidget *parent = 0);
    //----------------------------------------------------------------------------------------------------------------------
    /// @brief Destructor
    //----------------------------------------------------------------------------------------------------------------------
    ~MainWindow();
    //----------------------------------------------------------------------------------------------------------------------
private:
    //----------------------------------------------------------------------------------------------------------------------
    /// @brief Ui
    //----------------------------------------------------------------------------------------------------------------------
    Ui::MainWindow *ui;
    //----------------------------------------------------------------------------------------------------------------------
    /// @brief OpenGL widget to manage the scene
    //----------------------------------------------------------------------------------------------------------------------
    OpenGLWidget *m_openGLWidget;
    //----------------------------------------------------------------------------------------------------------------------
public slots:
    //----------------------------------------------------------------------------------------------------------------------
    /// @brief Sets the choppiness value displayed on the line edit
    //----------------------------------------------------------------------------------------------------------------------
    void choppinessValue(int _value);
    //----------------------------------------------------------------------------------------------------------------------
    /// @brief Sets the time value displayed on the line edit
    //----------------------------------------------------------------------------------------------------------------------
    void simulationTime(float _time);
    //----------------------------------------------------------------------------------------------------------------------
    /// @brief Sets the background colour on the base colour push button
    //----------------------------------------------------------------------------------------------------------------------
    void changeSeaBaseColourBtn();
    //----------------------------------------------------------------------------------------------------------------------
    /// @brief the background colour on the top colour of the push button
    //----------------------------------------------------------------------------------------------------------------------
    void changeSeaTopColourBtn();
    //----------------------------------------------------------------------------------------------------------------------
};

#endif // MAINWINDOW_H
/*@}*/
